common/init.sls

```diff
@@ -1,11 +1,38 @@
 {% from tpldir ~ '/map.jinja' import config with context %}
 
 {% if grains['kernel'] != 'Darwin' %}
-python-pkgs:
+# Ubuntu has python2 as default python
+python2:
   pkg.installed:
     - pkgs:
-      - python-pip
+      - python
       - python-dev
+
+python3:
+  pkg.installed:
+    - pkgs:
+      - python3
+
+# Ensure pip is default by purging pip3
+pip:
+  pkg.installed:
+    - pkgs:
+      - python-pip
+    - reload_modules: True
+
+pip3:
+  pkg.purged:
+    - pkgs:
+      - python3-pip
+
+# Virtualenv package creates virtualenv and virtualenv-3.4 executables
```

Comment:
I just tested this in a fresh vagrant box. It started out without pip or virtualenv (any version). I ran
1. `sudo apt-get install python-pip`
2. `sudo pip install virtualenv`
This created `/usr/local/bin/virtualenv` and `/usr/local/bin/virtualenv-3.4` executables, the latter of which works fine for the homu venv - I peeked inside the vagrant box and the homu venv had python3 and pip3.

bin_env is used when you want to run pip inside a virtualenv, such as in `homu/init.sls`, so I'm not sure what you are proposing with the two keys. Could you please clarify?